File: java/chart/src/ruijie/whistle/lightapp/request/Request.java
package ruijie.whistle.lightapp.request;

import org.dom4j.Element;

/**
* Note:抽象类,HTTP请求报文<br>
*/ 
public abstract class Request {
	public String FromUsername = null;
	public String MsgId = null;
	public String ToUsername;
	public String CreateTime;
	public Element Doc;

	/**
	* Note:获取发送者学号 <br> <br>
	* 	public String getFromStudentNumber();
	* @return 返回发送者学号
	*/ 
	public String getMsgId() {
		return Doc.elementText("msgid");
	}

	/**
	* Note:获取消息类型 <br> <br>
	* 	public String getType();
	* @return 消息类型
	*/ 
	public String getType() {
		return Doc.elementText("type");
	}

	/**
	* Note:获取发送者学号 <br> <br>
	* 	public String getFromStudentNumber();
	* @return 返回发送者学号
	*/ 
	public String getFromStudentNumber() {
		return Doc.elementText("fromstudentnumber");
	}

	/**
	* Note:获取发送者用户名 <br> <br>
	* 	public String getFromUsername();
	* @return 返回发送者用户名
	*/ 
	public String getFromUsername() {
		return Doc.elementText("fromusername");
	}

	/**
	* Note:获取接收者学号 <br> <br>
	* 	public String getToUsername();
	* @return 返回接收者学号
	*/ 
	public String getToUsername() {
		return Doc.elementText("tousername");
	}

	/**
	* Note:获取创建时间 <br> <br>
	* 	public long getCreateTime();
	* @return 创建时间 
	*/ 
	public long getCreateTime() {
		return Long.parseLong(Doc.elementText("createtime"));
	}

	/**
	* Note:获取XML报文 <br> <br>
	* 	protected Element getDoc()
	* @return 返回XML报文
	*/ 
	protected Element getDoc() {
		return Doc;
	}

	/**
	* Note:构造方法 <br> <br>
	* 	public Request(Element Doc);
	*/ 
	public Request(Element Doc) {
		this.Doc = Doc;
	}
}

File: php/chart/src/request/request.php
<?php

class Request {

    public $Doc;  

    function __construct($xml) {
        $this->Doc = $xml;
    }

    public function getDoc() {
        return $this->Doc;
    }

    public function getFromUsername() {
        return $this->Doc->fromusername;
    }

    public function getMsgId() {
        return $this->Doc->msgid;
    }

    public function getToUsername() {
        return $this->Doc->tousername;
    }

    public function getCreateTime() {
        return $this->Doc->createtime;
    }

    public function getType() {
        return $this->Doc->msgtype;
    }

    public function getFromStudentNumber() {
        return $this->Doc->fromstudentnumber;
    }

}

class HelloRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

    public function getContent() {
        return $this->Doc->content;
    }

}

class TextRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

    public function getContent() {
        return $this->Doc->content;
    }

}

class VoiceRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

}

class ImageRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

    public function getPicUrl() {
        return $this->Doc->picurl;
    }

}

class LocationRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

    public function getLabel() {
        return $this->Doc->label;
    }

    public function getScale() {
        return $this->Doc->scale;
    }

    public function getLocation_X() {
        return $this->Doc->location_x;
    }

    public function getLocation_Y() {
        return $this->Doc->location_y;
    }

}

class LinkRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

    public function getTitle() {
        return $this->Doc->title;
    }

    public function getDescription() {
        return $this->Doc->description;
    }

    public function getUrl() {
        return $this->Doc->url;
    }

}

class EventRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

    public function getEvent() {
        return $this->Doc->event;
    }

    public function getEventKey() {
        return $this->Doc->eventkey;
    }

}

class UnknownRequest extends Request {

    function __construct($xml) {
        parent::__construct($xml);
    }

}
